File: Back/Pizzas/Pizzas.Core/Dtos/Update/UpdatePizzaDto.cs
using Pizzas.Core.Dtos.Read;
using Pizzas.Core.Enums;

namespace Pizzas.Core.Dtos.Update;

public class UpdatePizzaDto
{
    public string? Category { get; set; }
    public string? Name { get; set; } 
    public string? Description { get; set; }
    public decimal? Rating { get; set; }
    public string? ImageUrl { get; set; }
    public bool? Stock { get; set; }
    public bool? Top { get; set; }
    public PizzaSize? Size { get; set; }
    public ICollection<UpdateIngredientDto>? Ingredients { get; set; } = [];
    public ICollection<UpdatePizzaPriceDto>? Prices { get; set; }
}

File: Front/Admin/src/core/dtos/update/updatePizza.dto.ts
import type { PizzaSize } from "../../types/pizza.type";
import type { UpdateIngredientDto } from "./updateIngredient.dto";
import type { UpdatePizzaPriceDto } from "./updatePizzaPrice.dto";

export interface UpdatePizzaDto {
  name?: string;
  category?: string;
  description?: string;
  rating?: number;
  imageUrl?: string;
  stock?: boolean;
  top?: boolean;
  size?: PizzaSize;
  ingredients?: UpdateIngredientDto[];
  prices?: UpdatePizzaPriceDto[];
}
